File: app/src/main/java/com/plbear/iweight/base/AppManager.java
package com.plbear.iweight.base;

import android.app.Activity;

import java.util.Stack;

/**
 * Created by yanyongjun on 2018/6/11.
 */

public class AppManager {
    private static AppManager mInstance = null;
    private static Stack<Activity> mActivities = new Stack<>();

    public static AppManager getAppManager() {
        if (mInstance != null) {
            return mInstance;
        }
        mInstance = new AppManager();
        return mInstance;
    }

    private AppManager() {

    }

    public void addToStack(Activity activity) {
        mActivities.push(activity);
    }

    public void removeFromStack(Activity activity) {
        mActivities.remove(activity);
    }

    public void finishAllActivity() {
        while (!mActivities.isEmpty()) {
            Activity activity = mActivities.pop();
            activity.finish();
        }
    }
}

File: app/src/main/java/com/plbear/iweight/base/AppManager.kt
package com.plbear.iweight.base

import android.app.Activity
import java.util.*

/**
 * Created by yanyongjun on 2018/4/5.
 */
class AppManager {
    companion object {
        private var mInstance: AppManager? = null
        private var mActivities = Stack<Activity>()
        fun getAppManager(): AppManager {
            if (mInstance != null) {
                return mInstance!!
            }
            mInstance = AppManager()
            return mInstance!!
        }
    }

    fun addToStack(activity: Activity) {
        mActivities.push(activity)
    }

    fun removeFromStack(activity: Activity) {
        mActivities.remove(activity)
    }

    fun finshAllActivity() {
        while (!mActivities.isEmpty()) {
            var activity = mActivities.pop()
            activity.finish()
        }
    }
}